tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -27,7 +27,7 @@ public function testCharsetCollationWhenCreatingForeignRelations(): void
             self::markTestSkipped('This test is useful for all databases, but designed only for mysql.');
         }
 
-        if (method_exists(AbstractPlatform::class, 'getGuidExpression')) {
+        if (! class_exists(Visitor::class)) {
             self::markTestSkipped('Test valid for doctrine/dbal:3.x only.');
```

Comment:
This is a bit inaccurate: the scenario being tested doesn't depend on the Visitor and is valid for DBAL 4 as well. I'd rather have two different sets of assertions depending on whether the visitor exists or not.

Once the support for DBAL 3 is dropped, only one branch will be left. Otherwise, most likely this test will be mechanically removed as no longer relevant.

Reply:
Sure, I can work on that.

Reply:
Note that I'm only using `Visitor` in order to detect the version, I don't really care about that class.

Reply:
> Note that I'm only using `Visitor` in order to detect the version, I don't really care about that class.

You should be checking for `RemoveNamespacedAssets` instead since the result of the test does depend on whether it exists or not.